airbyte-ci/connectors/live-tests/src/live_tests/commons/connection_objects_retrieval.py

```diff
@@ -100,18 +108,24 @@ def get_connection_objects(
         retrieval_reason (Optional[str]): The reason to access the connection objects.
         fail_if_missing_objects (bool, optional): Whether to raise a ValueError if a required object is missing. Defaults to True.
         connector_image (Optional[str]): The image name for the connector under test.
+        auto_select_connection (bool, optional): Whether to automatically select a connection if no connection id is passed. Defaults to False.
+        selected_streams (Optional[Set[str]]): The set of selected streams to use when auto selecting a connection.
     Raises:
         click.UsageError: If a required object is missing for the command.
         click.UsageError: If a retrieval reason is missing when passing a connection id.
     Returns:
         ConnectionObjects: The connection objects values.
     """
+    if connection_id is None and not auto_select_connection:
+        raise ValueError("A connection id or auto_select_connection must be provided to retrieve the connection objects.")
+    if auto_select_connection and not connector_image:
+        raise ValueError("A connector image must be provided when using auto_select_connection.")
 
     custom_config = get_connector_config_from_path(custom_config_path) if custom_config_path else None
     custom_configured_catalog = get_configured_catalog_from_path(custom_configured_catalog_path) if custom_configured_catalog_path else None
```

Comment:
WDTY about also applying `selected_streams` to the custom configured catalog, so that a user can pass in a catalog and doesn't have to edit it in order to test only the selected streams?